### ISSUE TITLE
[Question] I'd like to sync certain objects stored locally. Do I walk right way choosing PFIncrementalStore?

### ISSUE THREAD

**Issue by kas-kad** [sbonami/PFIncrementalStore#54]
Hello, my application is kind of virtual wallet. It has class Wallet which can be local wallet or shared wallet (let's call it shwallet). All data related to wallet is stored locally (Core Data + SQLite). The data related to shwallet needs to be stored both locally and remotely (so that all shwallet members are able to add their transactions to the shwallet). So my goal is to sync local shwallet's transactions with remote shwallet's transactions. What would you suggest me? Thank you.

**Comment by kas-kad** [comment]
here is a quote from Core Data Programming Guide:

> A coordinator can only be associated with one managed object model. If you want to put different entities into different stores, you must partition your model entities by defining configurations within the managed object models

Does it means that I should create separate configuration of datamodel for shwallets and swhallet's transactions? Or simply reassign the wallet object with method of NSManagedObjectContext`- (void)assignObject:(id)object toPersistentStore:(NSPersistentStore *)store` will be enough? How I see the solution: when I'm dealing with shwallet I assign it to PFIncrementalStore and on the other hand I assign it to standart local store (NSSQLiteStoreType). Am I getting it right?

**Comment by kas-kad** [comment]
There is another way to separately working with multiple persistent stores - using separate contexts which attached to certain persistent store coordinator (by setting `persistentStoreCoordinator` property).

**Comment by Alydyn** [comment]
I'd point out that it's standard in Objective C to use full variable names rather than abbreviated ones.  `shwallet` should probably be `sharedWallet`.  

Sorry, that was really bugging me. :smile:

**Comment by Alydyn** [comment]
@purrrminator I think this is more of an app design question, rather than a PFIS question.  Generally speaking, what you want to accomplish can be done with Core Data, PFIncrementalStore, and a backend like Parse but there are many factors specific to your application that would affect the actual configuration of those components.

Having said that, at it's most basic PFIS is meant for exactly what you mention in your first comment: syncing a local Core Data store with a Parse.com backend.

**Comment by Alydyn** [comment]
@purrrminator Regarding your second post, the whole intention of PFIS (and all other `NSIncrementalStore` implementations such as `AFIncrementalStore`) is that you, as the developer, only interact with Core Data in your code.  The `IncrementalStore` takes care of keeping the data in sync with the designated remote back end.  There is no reason for you to worry about assigning objects to specific persistent stores.

**Comment by kas-kad** [comment]
> Regarding your second post, the whole intention of PFIS (and all other NSIncrementalStore implementations such as AFIncrementalStore) is that you, as the developer, only interact with Core Data in your code. The IncrementalStore takes care of keeping the data in sync with the designated remote back end. There is no reason for you to worry about assigning objects to specific persistent stores.

thats right but how can I separate data that should be syncronized from one that shouldn't? I mean is there any way to choose which objects have to be sent on Parse cloud and which have not?

**Comment by kas-kad** [comment]
Why do I need the separation? to save some money buying Parse services

**Comment by Alydyn** [comment]
Save money?  What do you mean?  Have you looked at the Parse.com pricing?  Unless you are going to have millions upon millions of data objects, you won't incur a charge.  And at that point you should probably be able to afford $50-$100/month for more storage.

**Comment by kas-kad** [comment]
Well actually my manager said that if we are able to decrease **API-request frequency**, we'd better do this. I have no idea how strong the user-loading will be :)

**Comment by Alydyn** [comment]
Interesting.  Well, they did just change their pricing scheme.  There is no charge for API requests at all now.

**Comment by kas-kad** [comment]
actual request limit pricing here 
![2014-05-14 20 08 52](https://cloud.githubusercontent.com/assets/1947169/2973887/13e63fe2-db82-11e3-9309-687dabc3772e.png)

**Comment by Alydyn** [comment]
That is BURST limit, how many requests it will handle per second. 

EDIT: I would bet it would be extremely rare to exceed the free 30/second, again unless you are handling millions of users.

**Comment by kas-kad** [comment]
I assume the more requests our app sends the higher burst will be while thousands of users are working at the same time. Isn't?

**Comment by Alydyn** [comment]
That really depends.  The old pricing model gave you a limit of 20/second for free, and 40/second for the Pro level.  I had them price out an Enterprise package at many thousands per month and it still only included 40/second (EDIT 2: I just double-checked and it was actually 100/second.  But it was THOUSANDS per month).
Look at your analytics.  I would be very surprised if you exceed 30.  And, if you do exceed, Parse only returns a time out error and it(EDIT: "it" means client) attempts to try again (up to 5 times per request I believe).

Try running some scripts doing some API'y stuff and see what kind of burst you reach.  And remember that most "batch" type operations are going to only count as 1 request.

**Comment by kas-kad** [comment]
Well I really hope you are right. Becaue I really don't want to mess with all that core-data stuff :+1:
